test/parallel/test-worker-relative-path-double-dot.js

```diff
@@ -8,9 +8,15 @@ if (isMainThread) {
   const cwdName = path.relative('../', '.');
   const relativePath = path.relative('.', __filename);
   const w = new Worker(path.join('..', cwdName, relativePath));
+  process.chdir('..');
   w.on('message', common.mustCall((message) => {
-    assert.strictEqual(message, 'Hello, world!');
+    assert.strictEqual(message, `Hello, world! ${process.cwd()}`);
+    process.chdir('..');
   }));
 } else {
-  parentPort.postMessage('Hello, world!');
+  const cwd = process.cwd();
+  parentPort.postMessage(`Hello, world! ${cwd}`);
+  setTimeout(() => {
+    assert.notStrictEqual(process.cwd(), cwd);
+  }, 10);
```

Comment:
I think a timeout of 10 ms might end up being flaky. Can we do this through another message being passed to the worker?

Also, it might be a good idea to create a separate test file, rather than piggy-backing on an unrelated test…